### DISCUSSION TITLE
Sending to different emails

### DISCUSSION THREAD

**Discussion by JonthueM** [PHPMailer/PHPMailer#2887] [Q&A] [answered]
I notice that the address that is used for i.e. Fervent foodies is also used for The Conversation. I tried to clear it as you see below on Fervent foodies but php didn't like that lol!

```php
foreach($_POST['growgroups'] as $growgroups)
		{
			
			if ($growgroups === 'Fervent Foodies')
			{
				$mail->Subject = "Interested in Grow Group Fervent Foodies";
				$mail->addAddress("sitemaster@1usda.org");
				$mail->send();
				$mail->addAddress("");
			} 
			if ($growgroups === 'The Conversation')
			{
				$mail->Subject = "Interested in Grow Group The Conversation";
				$mail->addAddress('michel.bkyc@gmail.com');
				$mail->send();
			}
			if ($growgroups === 'Wednesday Night Bible Study')
			{
				$mail->Subject = "Interested in Grow Group Wedensay Night Bible Study";
				$mail->addAddress('michel.bkyc@gmail.com');
				$mail->send();
			} 
			if ($growgroups === 'Its Not About Me')
			{
				$mail->Subject = "Interested in Grow Group It's Not About Me";
				$mail->addAddress('michel.bkyc@gmail.com');
				$mail->send();
			} 
			if ($growgroups === 'Young Couples Mentorship')
			{
				$mail->Subject = "Interested in Grow Group Young Couples Mentorship";
				$mail->addAddress('sitemaster@1usda.org');
				$mail->send();
			} 
			if ($growgroups === 'Whats Next')
			{
				$mail->Subject = "Interested in Grow Group Whats New";
				$mail->addAddress("michel.bkyc@gmail.com");
				$mail->send();
			}  
		}
```

**Comment by Synchro** [comment, marked as answer]
There's a clue in the name of the method `addAddress` *adds* an address; it doesn't *set* the addresses the message is going to, and that list of recipients is retained for the life of the PHPMailer instance. So if you call `addAddress` repeatedly (even if you call `send` between them), it will send to all of those people. You can clear the list of addresses the message goes to by calling `clearAddresses()`. In your code, call `$mail->clearAddresses()` after each call to `$mail->send()` and it should work as you want.

**Reply by JonthueM** [reply]
TY Sooo much! I am new to php and mainly got on it to bypass a restricting framework. I notice that the more I add to this PHP file the slower it is to get a response to the end user, is there a way I can speed it up? I am planning on adding a few more layers of code to save the data either to SQL or google sheets @Synchro

**Reply by Synchro** [reply]
It will be the sending that is the slow part. The best thing you can do to improve that is to use a local mail server for relaying (so you don't have to worry about slow remote connections), and aside from that, your requirements have a lot in common with what's needed for mailing lists, in particular using `keepalive`, so have a read of [this wiki article](https://github.com/PHPMailer/PHPMailer/wiki/Sending-to-lists).

**Reply by JonthueM** [reply]
Will Since it is remote, will iframe workout?

**Comment by khan5501** [comment]
I want to use this code on [laptopswish](https://laptopswish.com/) but can you explain what changes are required it to start working?
<?php
$to = 'recipient@example.com';
$subject = 'Hello from PHP!';
$message = 'This is a test email sent using PHP.';
$headers = 'From: sender@example.com' . "\r\n" .
    'Reply-To: sender@example.com' . "\r\n" .
    'X-Mailer: PHP/' . phpversion();

if (mail($to, $subject, $message, $headers)) {
    echo 'Email sent successfully!';
} else {
    echo 'An error occurred while sending the email.';
}
?>

**Reply by Synchro** [reply]
Please don't comment on old, closed issues; it sends irrelevant notifications to everyone here. To use PHPMailer, you need to know how to use PHP to start with, and then read the wiki, examples and other docs provided here, search on Stack Overflow, and put together a script that does what you want (feel free to ask ChatGPT to help you). If you want to to send email from the contact form on your site, start with [the contact form example](https://github.com/PHPMailer/PHPMailer/blob/master/examples/contactform.phps). If you can't do it yourself, employ someone to help you.
